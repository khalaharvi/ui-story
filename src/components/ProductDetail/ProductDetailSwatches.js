import React from 'react';
import PropTypes from 'prop-types';

import { slugify } from 'utils';
import { cx } from './ProductDetail';

const ProductDetailSwatches = ({ title, items }) => (
  <div className={cx('swatches')}>
    <h2 className={cx('title')}>{title}</h2>
    <ul className={cx('swatches-list')}>
      {
        items.map(item => (
          <li key={`dh-swatch-${slugify(item.label)}`} className={cx('swatches-list-item')}>
            <figure>
              <img src={item.image} alt={item.label} />
              <figcaption>{item.label}</figcaption>
            </figure>
          </li>
        ))
      }
    </ul>
  </div>
);

ProductDetailSwatches.propTypes = {
  title: PropTypes.string,
  items: PropTypes.arrayOf(PropTypes.shape({
    image: PropTypes.string.isRequired,
    label: PropTypes.string
  })).isRequired
};

ProductDetailSwatches.defaultProps = {
  title: 'Available Finishes'
};

export default ProductDetailSwatches;
