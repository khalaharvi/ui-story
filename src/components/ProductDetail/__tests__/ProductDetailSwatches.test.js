import React from 'react';
import { shallow } from 'enzyme';

import ProductDetailSwatches from '../ProductDetailSwatches';

describe('<ProductDetailSwatches />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <ProductDetailSwatches
        items={[
          {
            label: 'Satin Brass',
            image: 'https://cdn.deringhall.com/swatch/image/50cdedfe-b5fe-4386-87cd-98c8c6af768c_satin.brass.jpg?auto=format&h=210&q=60&w=210'
          },
          {
            label: 'Antiqued Bronze',
            image: 'https://cdn.deringhall.com/swatch/image/afaf2371-2e28-45ed-80a1-a35a905328ba_antiqued.bronze.jpg?auto=format&h=210&q=60&w=210'
          },
          {
            label: 'Nickel',
            image: 'https://cdn.deringhall.com/swatch/image/e2162837-d502-4fcc-91a4-bc8aba8687fb_nickel.jpg?auto=format&h=210&q=60&w=210'
          },
          {
            label: 'Polished Brass',
            image: 'https://cdn.deringhall.com/swatch/image/8461198b-cb91-4d4e-a90b-b79ffd43bb2a_polished.brass.jpg?auto=format&h=210&q=60&w=210'
          },
          {
            label: 'Satin Brass',
            image: 'https://cdn.deringhall.com/swatch/image/50cdedfe-b5fe-4386-87cd-98c8c6af768c_satin.brass.jpg?auto=format&h=210&q=60&w=210'
          },
          {
            label: 'Antiqued Bronze',
            image: 'https://cdn.deringhall.com/swatch/image/afaf2371-2e28-45ed-80a1-a35a905328ba_antiqued.bronze.jpg?auto=format&h=210&q=60&w=210'
          },
          {
            label: 'Nickel',
            image: 'https://cdn.deringhall.com/swatch/image/e2162837-d502-4fcc-91a4-bc8aba8687fb_nickel.jpg?auto=format&h=210&q=60&w=210'
          },
          {
            label: 'Polished Brass',
            image: 'https://cdn.deringhall.com/swatch/image/8461198b-cb91-4d4e-a90b-b79ffd43bb2a_polished.brass.jpg?auto=format&h=210&q=60&w=210'
          },
          {
            label: 'Satin Brass',
            image: 'https://cdn.deringhall.com/swatch/image/50cdedfe-b5fe-4386-87cd-98c8c6af768c_satin.brass.jpg?auto=format&h=210&q=60&w=210'
          },
          {
            label: 'Antiqued Bronze',
            image: 'https://cdn.deringhall.com/swatch/image/afaf2371-2e28-45ed-80a1-a35a905328ba_antiqued.bronze.jpg?auto=format&h=210&q=60&w=210'
          },
          {
            label: 'Nickel',
            image: 'https://cdn.deringhall.com/swatch/image/e2162837-d502-4fcc-91a4-bc8aba8687fb_nickel.jpg?auto=format&h=210&q=60&w=210'
          },
          {
            label: 'Polished Brass',
            image: 'https://cdn.deringhall.com/swatch/image/8461198b-cb91-4d4e-a90b-b79ffd43bb2a_polished.brass.jpg?auto=format&h=210&q=60&w=210'
          }
        ]}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with custom title', () => {
    const wrapper = shallow(
      <ProductDetailSwatches
        title="Available Swatches"
        items={[
          {
            label: 'Satin Brass',
            image: 'https://cdn.deringhall.com/swatch/image/50cdedfe-b5fe-4386-87cd-98c8c6af768c_satin.brass.jpg?auto=format&h=210&q=60&w=210'
          },
          {
            label: 'Antiqued Bronze',
            image: 'https://cdn.deringhall.com/swatch/image/afaf2371-2e28-45ed-80a1-a35a905328ba_antiqued.bronze.jpg?auto=format&h=210&q=60&w=210'
          },
          {
            label: 'Nickel',
            image: 'https://cdn.deringhall.com/swatch/image/e2162837-d502-4fcc-91a4-bc8aba8687fb_nickel.jpg?auto=format&h=210&q=60&w=210'
          },
          {
            label: 'Polished Brass',
            image: 'https://cdn.deringhall.com/swatch/image/8461198b-cb91-4d4e-a90b-b79ffd43bb2a_polished.brass.jpg?auto=format&h=210&q=60&w=210'
          },
          {
            label: 'Satin Brass',
            image: 'https://cdn.deringhall.com/swatch/image/50cdedfe-b5fe-4386-87cd-98c8c6af768c_satin.brass.jpg?auto=format&h=210&q=60&w=210'
          },
          {
            label: 'Antiqued Bronze',
            image: 'https://cdn.deringhall.com/swatch/image/afaf2371-2e28-45ed-80a1-a35a905328ba_antiqued.bronze.jpg?auto=format&h=210&q=60&w=210'
          },
          {
            label: 'Nickel',
            image: 'https://cdn.deringhall.com/swatch/image/e2162837-d502-4fcc-91a4-bc8aba8687fb_nickel.jpg?auto=format&h=210&q=60&w=210'
          },
          {
            label: 'Polished Brass',
            image: 'https://cdn.deringhall.com/swatch/image/8461198b-cb91-4d4e-a90b-b79ffd43bb2a_polished.brass.jpg?auto=format&h=210&q=60&w=210'
          },
          {
            label: 'Satin Brass',
            image: 'https://cdn.deringhall.com/swatch/image/50cdedfe-b5fe-4386-87cd-98c8c6af768c_satin.brass.jpg?auto=format&h=210&q=60&w=210'
          },
          {
            label: 'Antiqued Bronze',
            image: 'https://cdn.deringhall.com/swatch/image/afaf2371-2e28-45ed-80a1-a35a905328ba_antiqued.bronze.jpg?auto=format&h=210&q=60&w=210'
          },
          {
            label: 'Nickel',
            image: 'https://cdn.deringhall.com/swatch/image/e2162837-d502-4fcc-91a4-bc8aba8687fb_nickel.jpg?auto=format&h=210&q=60&w=210'
          },
          {
            label: 'Polished Brass',
            image: 'https://cdn.deringhall.com/swatch/image/8461198b-cb91-4d4e-a90b-b79ffd43bb2a_polished.brass.jpg?auto=format&h=210&q=60&w=210'
          }
        ]}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
