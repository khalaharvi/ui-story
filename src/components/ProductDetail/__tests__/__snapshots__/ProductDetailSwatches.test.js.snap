// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductDetailSwatches /> renders correctly 1`] = `
<div
  className="dh-product-detail-swatches"
>
  <h2
    className="dh-product-detail-title"
  >
    Available Finishes
  </h2>
  <ul
    className="dh-product-detail-swatches-list"
  >
    <li
      className="dh-product-detail-swatches-list-item"
      key="dh-swatch-satin-brass"
    >
      <figure>
        <img
          alt="Satin Brass"
          src="https://cdn.deringhall.com/swatch/image/50cdedfe-b5fe-4386-87cd-98c8c6af768c_satin.brass.jpg?auto=format&h=210&q=60&w=210"
        />
        <figcaption>
          Satin Brass
        </figcaption>
      </figure>
    </li>
    <li
      className="dh-product-detail-swatches-list-item"
      key="dh-swatch-antiqued-bronze"
    >
      <figure>
        <img
          alt="Antiqued Bronze"
          src="https://cdn.deringhall.com/swatch/image/afaf2371-2e28-45ed-80a1-a35a905328ba_antiqued.bronze.jpg?auto=format&h=210&q=60&w=210"
        />
        <figcaption>
          Antiqued Bronze
        </figcaption>
      </figure>
    </li>
    <li
      className="dh-product-detail-swatches-list-item"
      key="dh-swatch-nickel"
    >
      <figure>
        <img
          alt="Nickel"
          src="https://cdn.deringhall.com/swatch/image/e2162837-d502-4fcc-91a4-bc8aba8687fb_nickel.jpg?auto=format&h=210&q=60&w=210"
        />
        <figcaption>
          Nickel
        </figcaption>
      </figure>
    </li>
    <li
      className="dh-product-detail-swatches-list-item"
      key="dh-swatch-polished-brass"
    >
      <figure>
        <img
          alt="Polished Brass"
          src="https://cdn.deringhall.com/swatch/image/8461198b-cb91-4d4e-a90b-b79ffd43bb2a_polished.brass.jpg?auto=format&h=210&q=60&w=210"
        />
        <figcaption>
          Polished Brass
        </figcaption>
      </figure>
    </li>
    <li
      className="dh-product-detail-swatches-list-item"
      key="dh-swatch-satin-brass"
    >
      <figure>
        <img
          alt="Satin Brass"
          src="https://cdn.deringhall.com/swatch/image/50cdedfe-b5fe-4386-87cd-98c8c6af768c_satin.brass.jpg?auto=format&h=210&q=60&w=210"
        />
        <figcaption>
          Satin Brass
        </figcaption>
      </figure>
    </li>
    <li
      className="dh-product-detail-swatches-list-item"
      key="dh-swatch-antiqued-bronze"
    >
      <figure>
        <img
          alt="Antiqued Bronze"
          src="https://cdn.deringhall.com/swatch/image/afaf2371-2e28-45ed-80a1-a35a905328ba_antiqued.bronze.jpg?auto=format&h=210&q=60&w=210"
        />
        <figcaption>
          Antiqued Bronze
        </figcaption>
      </figure>
    </li>
    <li
      className="dh-product-detail-swatches-list-item"
      key="dh-swatch-nickel"
    >
      <figure>
        <img
          alt="Nickel"
          src="https://cdn.deringhall.com/swatch/image/e2162837-d502-4fcc-91a4-bc8aba8687fb_nickel.jpg?auto=format&h=210&q=60&w=210"
        />
        <figcaption>
          Nickel
        </figcaption>
      </figure>
    </li>
    <li
      className="dh-product-detail-swatches-list-item"
      key="dh-swatch-polished-brass"
    >
      <figure>
        <img
          alt="Polished Brass"
          src="https://cdn.deringhall.com/swatch/image/8461198b-cb91-4d4e-a90b-b79ffd43bb2a_polished.brass.jpg?auto=format&h=210&q=60&w=210"
        />
        <figcaption>
          Polished Brass
        </figcaption>
      </figure>
    </li>
    <li
      className="dh-product-detail-swatches-list-item"
      key="dh-swatch-satin-brass"
    >
      <figure>
        <img
          alt="Satin Brass"
          src="https://cdn.deringhall.com/swatch/image/50cdedfe-b5fe-4386-87cd-98c8c6af768c_satin.brass.jpg?auto=format&h=210&q=60&w=210"
        />
        <figcaption>
          Satin Brass
        </figcaption>
      </figure>
    </li>
    <li
      className="dh-product-detail-swatches-list-item"
      key="dh-swatch-antiqued-bronze"
    >
      <figure>
        <img
          alt="Antiqued Bronze"
          src="https://cdn.deringhall.com/swatch/image/afaf2371-2e28-45ed-80a1-a35a905328ba_antiqued.bronze.jpg?auto=format&h=210&q=60&w=210"
        />
        <figcaption>
          Antiqued Bronze
        </figcaption>
      </figure>
    </li>
    <li
      className="dh-product-detail-swatches-list-item"
      key="dh-swatch-nickel"
    >
      <figure>
        <img
          alt="Nickel"
          src="https://cdn.deringhall.com/swatch/image/e2162837-d502-4fcc-91a4-bc8aba8687fb_nickel.jpg?auto=format&h=210&q=60&w=210"
        />
        <figcaption>
          Nickel
        </figcaption>
      </figure>
    </li>
    <li
      className="dh-product-detail-swatches-list-item"
      key="dh-swatch-polished-brass"
    >
      <figure>
        <img
          alt="Polished Brass"
          src="https://cdn.deringhall.com/swatch/image/8461198b-cb91-4d4e-a90b-b79ffd43bb2a_polished.brass.jpg?auto=format&h=210&q=60&w=210"
        />
        <figcaption>
          Polished Brass
        </figcaption>
      </figure>
    </li>
  </ul>
</div>
`;

exports[`<ProductDetailSwatches /> renders correctly with custom title 1`] = `
<div
  className="dh-product-detail-swatches"
>
  <h2
    className="dh-product-detail-title"
  >
    Available Swatches
  </h2>
  <ul
    className="dh-product-detail-swatches-list"
  >
    <li
      className="dh-product-detail-swatches-list-item"
      key="dh-swatch-satin-brass"
    >
      <figure>
        <img
          alt="Satin Brass"
          src="https://cdn.deringhall.com/swatch/image/50cdedfe-b5fe-4386-87cd-98c8c6af768c_satin.brass.jpg?auto=format&h=210&q=60&w=210"
        />
        <figcaption>
          Satin Brass
        </figcaption>
      </figure>
    </li>
    <li
      className="dh-product-detail-swatches-list-item"
      key="dh-swatch-antiqued-bronze"
    >
      <figure>
        <img
          alt="Antiqued Bronze"
          src="https://cdn.deringhall.com/swatch/image/afaf2371-2e28-45ed-80a1-a35a905328ba_antiqued.bronze.jpg?auto=format&h=210&q=60&w=210"
        />
        <figcaption>
          Antiqued Bronze
        </figcaption>
      </figure>
    </li>
    <li
      className="dh-product-detail-swatches-list-item"
      key="dh-swatch-nickel"
    >
      <figure>
        <img
          alt="Nickel"
          src="https://cdn.deringhall.com/swatch/image/e2162837-d502-4fcc-91a4-bc8aba8687fb_nickel.jpg?auto=format&h=210&q=60&w=210"
        />
        <figcaption>
          Nickel
        </figcaption>
      </figure>
    </li>
    <li
      className="dh-product-detail-swatches-list-item"
      key="dh-swatch-polished-brass"
    >
      <figure>
        <img
          alt="Polished Brass"
          src="https://cdn.deringhall.com/swatch/image/8461198b-cb91-4d4e-a90b-b79ffd43bb2a_polished.brass.jpg?auto=format&h=210&q=60&w=210"
        />
        <figcaption>
          Polished Brass
        </figcaption>
      </figure>
    </li>
    <li
      className="dh-product-detail-swatches-list-item"
      key="dh-swatch-satin-brass"
    >
      <figure>
        <img
          alt="Satin Brass"
          src="https://cdn.deringhall.com/swatch/image/50cdedfe-b5fe-4386-87cd-98c8c6af768c_satin.brass.jpg?auto=format&h=210&q=60&w=210"
        />
        <figcaption>
          Satin Brass
        </figcaption>
      </figure>
    </li>
    <li
      className="dh-product-detail-swatches-list-item"
      key="dh-swatch-antiqued-bronze"
    >
      <figure>
        <img
          alt="Antiqued Bronze"
          src="https://cdn.deringhall.com/swatch/image/afaf2371-2e28-45ed-80a1-a35a905328ba_antiqued.bronze.jpg?auto=format&h=210&q=60&w=210"
        />
        <figcaption>
          Antiqued Bronze
        </figcaption>
      </figure>
    </li>
    <li
      className="dh-product-detail-swatches-list-item"
      key="dh-swatch-nickel"
    >
      <figure>
        <img
          alt="Nickel"
          src="https://cdn.deringhall.com/swatch/image/e2162837-d502-4fcc-91a4-bc8aba8687fb_nickel.jpg?auto=format&h=210&q=60&w=210"
        />
        <figcaption>
          Nickel
        </figcaption>
      </figure>
    </li>
    <li
      className="dh-product-detail-swatches-list-item"
      key="dh-swatch-polished-brass"
    >
      <figure>
        <img
          alt="Polished Brass"
          src="https://cdn.deringhall.com/swatch/image/8461198b-cb91-4d4e-a90b-b79ffd43bb2a_polished.brass.jpg?auto=format&h=210&q=60&w=210"
        />
        <figcaption>
          Polished Brass
        </figcaption>
      </figure>
    </li>
    <li
      className="dh-product-detail-swatches-list-item"
      key="dh-swatch-satin-brass"
    >
      <figure>
        <img
          alt="Satin Brass"
          src="https://cdn.deringhall.com/swatch/image/50cdedfe-b5fe-4386-87cd-98c8c6af768c_satin.brass.jpg?auto=format&h=210&q=60&w=210"
        />
        <figcaption>
          Satin Brass
        </figcaption>
      </figure>
    </li>
    <li
      className="dh-product-detail-swatches-list-item"
      key="dh-swatch-antiqued-bronze"
    >
      <figure>
        <img
          alt="Antiqued Bronze"
          src="https://cdn.deringhall.com/swatch/image/afaf2371-2e28-45ed-80a1-a35a905328ba_antiqued.bronze.jpg?auto=format&h=210&q=60&w=210"
        />
        <figcaption>
          Antiqued Bronze
        </figcaption>
      </figure>
    </li>
    <li
      className="dh-product-detail-swatches-list-item"
      key="dh-swatch-nickel"
    >
      <figure>
        <img
          alt="Nickel"
          src="https://cdn.deringhall.com/swatch/image/e2162837-d502-4fcc-91a4-bc8aba8687fb_nickel.jpg?auto=format&h=210&q=60&w=210"
        />
        <figcaption>
          Nickel
        </figcaption>
      </figure>
    </li>
    <li
      className="dh-product-detail-swatches-list-item"
      key="dh-swatch-polished-brass"
    >
      <figure>
        <img
          alt="Polished Brass"
          src="https://cdn.deringhall.com/swatch/image/8461198b-cb91-4d4e-a90b-b79ffd43bb2a_polished.brass.jpg?auto=format&h=210&q=60&w=210"
        />
        <figcaption>
          Polished Brass
        </figcaption>
      </figure>
    </li>
  </ul>
</div>
`;
